packages/cinterop/src/jvmCommon/kotlin/io/realm/interop/RealmInterop.kt

```diff
@@ -22,6 +22,14 @@ package io.realm.interop
 private val INVALID_CLASS_KEY: Long by lazy { realmc.getRLM_INVALID_CLASS_KEY() }
 private val INVALID_PROPERTY_KEY: Long by lazy { realmc.getRLM_INVALID_PROPERTY_KEY() }
 
+actual class RealmValue private constructor(val _v: Any?, val _n: realm_value_t?)  {
```

Comment:
It would be doable to just expect/actual-ing the internal type, but we would end up needing to expose it as part of the `RealmInterop` interface, thus leaking the internal type through the API. 

Further, I think that it would be easier to handle any platform memory handling if we did the full implementation in the platform specific code. 

I tried to sketch it on native too, but it requires us to handle allocation of not only the actual struct but also the related memory, i.e. strings and binary data. As there is no need for that at the moment since it is a pure input parameter, I just ended up passing it as `Any?`. This way I could do the conversion to `realm_value_t` inside the specific `realm_object_create_with_primary_key` methods by existing mem-scoped methods for conversion.